src/vis/vis-view.js

```diff
@@ -33,6 +33,8 @@ var Vis = View.extend({
 
     this.model.once('load', this.render, this);
     this.model.on('invalidateSize', this._invalidateSize, this);
+    this.model.overlaysCollection.on('add remove', this._addOverlays, this);
```

Comment:
I did it that way because _addOverlays already cleans the overlays: https://github.com/CartoDB/cartodb.js/blob/v4/src/vis/vis-view.js#L224-L227